ratis-netty/src/main/java/org/apache/ratis/netty/server/DataStreamManagement.java

```diff
@@ -439,7 +448,7 @@ private void readImpl(DataStreamRequestByteBuf request, ChannelHandlerContext ct
         }, requestExecutor)).whenComplete((v, exception) -> {
       try {
         if (exception != null) {
-          streams.remove(key);
+          removeDataStream(request);
```

Comment:
We should pass `info` in this case since it could be already removed from `streams` at https://github.com/apache/ratis/blob/d41d0930c49c4cfcd8cd709519329e22fbbe61b4/ratis-netty/src/main/java/org/apache/ratis/netty/server/DataStreamManagement.java#L401

Reply:
Yes, I modify it